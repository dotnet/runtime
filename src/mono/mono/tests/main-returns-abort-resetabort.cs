
using System;
using System.Threading;

public class foo {
	public static void Main() {
		Thread thr=new Thread(new ThreadStart(foo.thread));
		thr.Start();
		Thread.Sleep(600);
		Console.WriteLine("Aborting child thread");
		thr.Abort();
		Console.WriteLine("Main thread returns");
	}

	public static void thread() {
		try {
			Console.WriteLine("Thread running 1");
			Thread.Sleep(500);
		} catch(ThreadAbortException) {
			Thread.ResetAbort();
			Console.WriteLine("Abort reset! 1");
		} finally {
			Console.WriteLine("ThreadAbortException finally 1");
		}
		try {
			Console.WriteLine("Thread running 2");
			Thread.Sleep(500);
		} catch(ThreadAbortException) {
			Thread.ResetAbort();
			Console.WriteLine("Abort reset! 2");
		} finally {
			Console.WriteLine("ThreadAbortException finally 2");
		}
		try {
			Console.WriteLine("Thread running 3");
			Thread.Sleep(500);
		} catch(ThreadAbortException) {
			Thread.ResetAbort();
			Console.WriteLine("Abort reset! 3");
		} finally {
			Console.WriteLine("ThreadAbortException finally 3");
		}
		try {
			Console.WriteLine("Thread running 4");
			Thread.Sleep(500);
		} catch(ThreadAbortException) {
			Thread.ResetAbort();
			Console.WriteLine("Abort reset! 4");
		} finally {
			Console.WriteLine("ThreadAbortException finally 4");
		}
		try {
			Console.WriteLine("Thread running 5");
		} catch(ThreadAbortException) {
			Thread.ResetAbort();
			Console.WriteLine("Abort reset! 5");
		} finally {
			Console.WriteLine("ThreadAbortException finally 5");
		}
	}
}

