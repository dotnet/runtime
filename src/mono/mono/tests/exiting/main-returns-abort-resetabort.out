Thread running
ThreadAbortException finally
Thread running
Aborting child thread
Main thread returns
Abort reset!
ThreadAbortException finally
Thread running
ThreadAbortException finally
Thread running
ThreadAbortException finally
Thread running
ThreadAbortException finally
